File: frontend/src/mateu/shared/apiClients/dtos/Menu.ts
import {MenuType} from "./MenuType";

export default interface Menu {

    type: MenuType;

    id: string;

    icon: string;

    caption: string;

    journeyTypeId: string | undefined;

    useJourneyTypeForId: boolean | undefined;

    submenus: Menu[] | null;

}

File: backend/dtos/src/main/java/io/mateu/dtos/Menu.java
package io.mateu.dtos;

import com.fasterxml.jackson.annotation.JsonIgnore;
import java.util.Collections;
import java.util.List;
import lombok.*;

public record Menu(
    MenuType type,
    String icon,
    String caption,
    String journeyTypeId,
    List<Menu> submenus,
    @JsonIgnore int order) {

  public Menu {
    submenus = Collections.unmodifiableList(submenus);
  }

  @Override
  public List<Menu> submenus() {
    return Collections.unmodifiableList(submenus);
  }
}
